cgi-bin/DW/Setting/StickyEntry.pm

```diff
@@ -30,17 +29,40 @@ sub option {
     my $key = $class->pkgkey;
     my $ret;
 
-    $ret .= LJ::html_text({
-        name  => "${key}stickyid",
-        id    => "${key}stickyid",
-        class => "text",
-        value => $errs ? $class->get_arg( $args, "stickyid" ) : $u->sticky_entry,
-        size  => 30,
-        maxlength => 100,
-    });
+    my @stickies = $u->sticky_entries;
+    my $username = $u->user;
 
+    foreach my $i ( 1...$u->count_max_stickies ) {
+        my $url = "";
+        my $placeholder = "";
+        my $e = $stickies[$i - 1];
+        if ( $e ) {
+            $url = $e->url;
+        } else {
+            $placeholder = $u->journal_base . "/1234.html" if ( $i == 1 );
+        }
+        my $textentry = $errs ? $class->get_arg( $args, "stickyid${i}" ) : $url;
+
+        $ret .= "<label for='${key}stickyid${i}'>" . $class->ml( 'setting.stickyentryi.label', { stickyid => $i } ) . " </label>";
+        $ret .= LJ::html_text({
+            name  => "${key}stickyid${i}",
+            id    => "${key}stickyid${i}",
+            class => "text",
+            value => "$textentry",
+            placeholder => $textentry ? "" : $placeholder,
+            size  => 50,
+            maxlength => 100,
+        });
+        $ret .= q{ - <a href='} . $e->url . q{'>} . ( $e->subject_html || "(no subject)" ) . q{</a>} if $e;
+        $ret .= "<br>";
+
+    }
+
+    # returns an error if any of the stickies are incorrectly formatted.
     my $errdiv = $class->errdiv( $errs, "stickyid" );
-    $ret .= "<br />$errdiv" if $errdiv;
+    $ret .= "$errdiv<br>" if $errdiv;
```

Comment:
Mind blown.

--
Mark Smith mark@qq.is


On Mon, Feb 2, 2015, at 10:23 PM, Afuna wrote:
> In cgi-bin/DW/Setting/StickyEntry.pm[1]:


> >      my $errdiv = $class->errdiv( $errs, "stickyid" );
> -    $ret .= "<br />$errdiv" if $errdiv;
> +    $ret .= "$errdiv<br>" if $errdiv; That's true for XHTML, but it's
>      optional on HTML5 (which is what we're aiming for now).


> — Reply to this email directly or view it on GitHub[2].

>


Links:

  1. https://github.com/dreamwidth/dw-free/pull/1182#discussion_r23985084
  2. https://github.com/dreamwidth/dw-free/pull/1182/files#r23985084